src/Basescape/CraftSoldiersState.cpp

```diff
@@ -297,6 +297,7 @@ void CraftSoldiersState::init()
 void CraftSoldiersState::lstItemsLeftArrowClick(Action *action)
 {
 	unsigned int row = _lstSoldiers->getSelectedRow();
+    size_t numSoldiers = _base->getSoldiers()->size();
```

Comment:
This is already done in CraftSoldiersState::lstSoldiersMousePress(), same as in vanilla.
Other 4 classes should probably have it implemented the same way.

Reply:
You're right. Looking at that, it seems strange that the ordering handlers are separated in three different functions.

For CraftArmorState I tried a different approach: https://pastebin.com/jhW8zyr9

I think this is a better way because 1) it combines all arrow-related logic into one handler function and 2) the sanity check for ordering past the list bounds is handled in moveSoldierUp() and moveSoldierDown(), which is where it belongs.

The problem though is that implementing it like this would change code from OXC and OXCE, which could create merging issues later down the line.